kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/util/Metrics.java

```diff
@@ -14,47 +15,73 @@
+    public static final String KROXYLICIOUS_UPSTREAM_ERRORS = KROXYLICIOUS_UPSTREAM + "errors";
 
-    private static final String KROXYLICIOUS_INBOUND_DOWNSTREAM_DECODED_MESSAGES = "kroxylicious_inbound_downstream_decoded_messages";
+    public static final String KROXYLICIOUS_INBOUND_DOWNSTREAM_DECODED_MESSAGES = "kroxylicious_inbound_downstream_decoded_messages";
 
-    private static final String KROXYLICIOUS_PAYLOAD_SIZE_BYTES = "kroxylicious_payload_size_bytes";
+    public static final String KROXYLICIOUS_PAYLOAD_SIZE_BYTES = "kroxylicious_payload_size_bytes";
 
     private static final String FLOWING_TAG = "flowing";
+    public static final String VIRTUAL_CLUSTER_TAG = "virtualCluster";
 
-    private static final Tag FLOWING_UPSTREAM = Tag.of(FLOWING_TAG, "upstream");
+    public static final Tag FLOWING_DOWNSTREAM = Tag.of(FLOWING_TAG, "downstream");
 
-    private static final Tag FLOWING_DOWNSTREAM = Tag.of(FLOWING_TAG, "downstream");
+    @NonNull
+    public static Counter taggedCounter(String counterName, List<Tag> tags) {
+        return counter(counterName, tags);
+    }
 
     public static Counter inboundDownstreamMessagesCounter() {
-        return counter(KROXYLICIOUS_INBOUND_DOWNSTREAM_MESSAGES, List.of(FLOWING_DOWNSTREAM));
+        return taggedCounter(KROXYLICIOUS_INBOUND_DOWNSTREAM_MESSAGES, List.of(FLOWING_DOWNSTREAM));
     }
 
     public static Counter inboundDownstreamDecodedMessagesCounter() {
-        return counter(KROXYLICIOUS_INBOUND_DOWNSTREAM_DECODED_MESSAGES, List.of(FLOWING_DOWNSTREAM));
+        return taggedCounter(KROXYLICIOUS_INBOUND_DOWNSTREAM_DECODED_MESSAGES, List.of(FLOWING_DOWNSTREAM));
     }
 
     public static DistributionSummary payloadSizeBytesUpstreamSummary(ApiKeys apiKey, short apiVersion) {
-        return payloadSizeBytesSummary(apiKey, apiVersion, FLOWING_UPSTREAM);
+        return payloadSizeBytesSummary(apiKey, apiVersion, "upstream");
     }
 
     public static DistributionSummary payloadSizeBytesDownstreamSummary(ApiKeys apiKey, short apiVersion) {
-        return payloadSizeBytesSummary(apiKey, apiVersion, FLOWING_DOWNSTREAM);
+        return payloadSizeBytesSummary(apiKey, apiVersion, "downstream");
     }
 
-    private static DistributionSummary payloadSizeBytesSummary(ApiKeys apiKey, short apiVersion, Tag flowing) {
-        List<Tag> tags = List.of(
-                Tag.of("ApiKey", apiKey.name()),
-                Tag.of("ApiVersion", String.valueOf(apiVersion)),
-                flowing);
+    private static DistributionSummary payloadSizeBytesSummary(ApiKeys apiKey, short apiVersion, String flowing) {
+        List<Tag> tags = tags(
+                "ApiKey", apiKey.name(),
+                "ApiVersion", String.valueOf(apiVersion),
+                FLOWING_TAG, flowing);
         return summary(KROXYLICIOUS_PAYLOAD_SIZE_BYTES, tags);
     }
 
-}
+    public static List<Tag> tags(String... tagsAndValues) {
```

Comment:
Perhaps we should do like `Map.of()` and have overloaded 2- and 4-parameter methods so that messing up the commonest cases (of one or two tags) is a compile time error.